tests/models/llava/test_modeling_llava.py

```diff
@@ -247,20 +247,20 @@ def test_small_model_integration_test_llama_batched(self):
         model_id = "llava-hf/llava-1.5-7b-hf"
 
         model = LlavaForConditionalGeneration.from_pretrained("llava-hf/llava-1.5-7b-hf", load_in_4bit=True)
-        processor = AutoProcessor.from_pretrained(model_id, pad_token="<pad>")
+        processor = AutoProcessor.from_pretrained(model_id)
 
         prompts = [
             "USER: <image>\nWhat are the things I should be cautious about when I visit this place? What should I bring with me?\nASSISTANT:",
-            "USER: <image>\nWhat is this?\nASSISTANT: Two cats lying on a bed!\nUSER: <image>\nAnd this?\nASSISTANT:",
+            "USER: <image>\nWhat is this?\nASSISTANT:",
```

Comment:
@amyeroberts here there is a regression issue with SDPA support of Llava; when users perform batched generation with different number of images in prompts, the model output gibberish. I suspect this is related to the fact SDPA might not support arbitrary attention masks.

Therefore I adapted the test here to make sure we do perform batched generation with SDPA with the same number of images per prompt and I added a regression test that test the previous behaviour by loading the model with `attn_implementation="eager"`. 

I also adapted expected values with the ones we should get with T4s, which are used in our CIs. The original values were obtained on an A100, which led to failures currently. 

I am happy to break down this into multiple PRs, but I thought we could make all the fixes in a single PR

Reply:
Nice 🔥  - I think it's fine to keep both here

Reply:
Ok perfect then!